src/executables/sysrepo-plugind.c

```diff
@@ -438,17 +438,23 @@ main(int argc, char **argv)
     sr_session_ctx_t *sess = NULL;
     sr_log_level_t log_level = SR_LL_ERR;
     int plugin_count = 0, i, r, rc = EXIT_FAILURE, opt, debug = 0;
+    int pidfd = -1;
+    char pid[8] = { 0 };
```

Comment:
This is fragile. On Linux, the upper bound on the maximal PID values [has been increased before](https://lwn.net/Articles/10424/), and the fact that it's currently limited to 4M, and therefore will *currently* fit into seven digits plus the trailing `NULL` is just an implementation detail.

Reply:
It looks like I forgot to mention that I suggest to use `asprintf()` since it's something that's already used throughout the code.